README.md
# Msm

This project was generated with [Angular CLI](https://github.com/angular/angular-cli) version 9.1.7.

## Development server

Run `ng serve` for a dev server. Navigate to `http://localhost:4200/`. The app will automatically reload if you change any of the source files.

## Code scaffolding

Run `ng generate component component-name` to generate a new component. You can also use `ng generate directive|pipe|service|class|guard|interface|enum|module`.

## Build

Run `ng build` to build the project. The build artifacts will be stored in the `dist/` directory. Use the `--prod` flag for a production build.

## Running unit tests

Run `ng test` to execute the unit tests via [Karma](https://karma-runner.github.io).

## Running end-to-end tests

Run `ng e2e` to execute the end-to-end tests via [Protractor](http://www.protractortest.org/).

## Further help

To get more help on the Angular CLI use `ng help` or go check out the [Angular CLI README](https://github.com/angular/angular-cli/blob/master/README.md).


changelog
**MSM Xtended XQ Release-v8.0 Changelogs**
* Fixed DC Dimming
* Added DC Dimming QS Tile
* Added ringer mode checking for camera sound effect
* FOD: Added dispparam handling mechanism as fallback
* Fixed A2DP when offload is disabled 
* Improved FOD on AOD
* Enabled LED Notifications
* Fixed MultiUser Section FC
* Switched to April Crosshatch build fingerprint 
* Switched to Latest Immensity Kernel

**MSM Xtended XQ Release-v7.0 Changelogs**

- Updated Coral Build Fingerprint to March Security Patch
- Updated Anx Cam to latest
- Switched to Latest Immensity
- Enable vibration multiple intensities
* Activates some options on Settings>accessibility>vibration
- Used USB audio policy configuration from AOSP
* This can greatly improve audio quality when an external
  USB DAC is connected
- Fix dark theme issues
* This brings SwitchBar widget inline with 10 and fixes the
  theme issues when the dark mode is enabled
- Updated vibrator configurations
- Removed Ambient Display from Display Settings
- Fixed the shitty HD icon in Statusbar
- Fixed Ambient FC in Xtensions
- Defined My Name Tag in About section ðŸ˜Ž


**MSM Xtended XQ Release-v6.0 Changelogs**

* Added latest los fod changes
* Added Raise to wake feature
* Switched to F1XY Kernel (as per my liking Â¯\_(ãƒ„)_/Â¯)
* Enabled AOSP Bluetooth Hal v2
* Updated Coral's build fingerprint to Feb 2020 security patch
* Fixed Ambient in Decor Room
* Fixed Mi Sound Enhancer
* Fixed Popup Sound selection
* Fixed safetynet from previous test build


src/app/app.module.ts
import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { MatCardModule } from '@angular/material/card';
import { MatTableModule } from '@angular/material/table';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatPaginatorModule } from '@angular/material/paginator';
import { MatInputModule } from '@angular/material/input';
import { MatIconModule } from '@angular/material/icon';
// import { MatInputModule } from '@angular/material';
// import { MatInputModule } from '@angular/material';


@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    MatCardModule,
    MatFormFieldModule,
    MatInputModule,
    MatPaginatorModule,
    MatIconModule,
    BrowserAnimationsModule,
    MatTableModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }


src/app/devices/table.component.ts
import { Component } from '@angular/core';
import { DataSource } from '@angular/cdk/collections';
import { Observable, of } from 'rxjs';
import { animate, state, style, transition, trigger } from '@angular/animations';

@Component({
  selector: 'app-table',
  templateUrl: './table.component.html',
  styleUrls: ['./table.component.scss'],
  animations: [
    trigger('detailExpand', [
      state('collapsed', style({ height: '0px', minHeight: '0', visibility: 'hidden' })),
      state('expanded', style({ height: '*', visibility: 'visible' })),
      transition('expanded <=> collapsed', animate('225ms cubic-bezier(0.4, 0.0, 0.2, 1)')),
    ]),
  ],
})
export class TableComponent {
  displayedColumns = ['position', 'name', 'weight'];
  dataSource = new ExampleDataSource();

  isExpansionDetailRow = (i: number, row: Object) => row.hasOwnProperty('detailRow');
  expandedElement: any;
}

export interface Element {
  name: string;
  position: number;
  weight: number;
  symbol: string;
}

const data: Element[] = [
  { position: 1, name: 'Hydrogen', weight: 1.0079, symbol: 'H' },
  { position: 2, name: 'Helium', weight: 4.0026, symbol: 'He' },
  { position: 3, name: 'Lithium', weight: 6.941, symbol: 'Li' },
  { position: 4, name: 'Beryllium', weight: 9.0122, symbol: 'Be' },
  { position: 5, name: 'Boron', weight: 10.811, symbol: 'B' },
  { position: 6, name: 'Carbon', weight: 12.0107, symbol: 'C' },
  { position: 7, name: 'Nitrogen', weight: 14.0067, symbol: 'N' },
  { position: 8, name: 'Oxygen', weight: 15.9994, symbol: 'O' },
  { position: 9, name: 'Fluorine', weight: 18.9984, symbol: 'F' },
  { position: 10, name: 'Neon', weight: 20.1797, symbol: 'Ne' },
  { position: 11, name: 'Sodium', weight: 22.9897, symbol: 'Na' },
  { position: 12, name: 'Magnesium', weight: 24.305, symbol: 'Mg' },
  { position: 13, name: 'Aluminum', weight: 26.9815, symbol: 'Al' },
  { position: 14, name: 'Silicon', weight: 28.0855, symbol: 'Si' },
  { position: 15, name: 'Phosphorus', weight: 30.9738, symbol: 'P' },
  { position: 16, name: 'Sulfur', weight: 32.065, symbol: 'S' },
  { position: 17, name: 'Chlorine', weight: 35.453, symbol: 'Cl' },
  { position: 18, name: 'Argon', weight: 39.948, symbol: 'Ar' },
  { position: 19, name: 'Potassium', weight: 39.0983, symbol: 'K' },
  { position: 20, name: 'Calcium', weight: 40.078, symbol: 'Ca' },
];

/**
 * Data source to provide what data should be rendered in the table. The observable provided
 * in connect should emit exactly the data that should be rendered by the table. If the data is
 * altered, the observable should emit that new set of data on the stream. In our case here,
 * we return a stream that contains only one set of data that doesn't change.
 */
export class ExampleDataSource extends DataSource<any> {
  /** Connect function called by the table to retrieve one stream containing the data to render. */
  connect(): Observable<Element[]> {
    const rows = [];
    data.forEach(element => rows.push(element, { detailRow: true, element }));
    console.log(rows);
    return of(rows);
  }

  disconnect() { }
}


/**  Copyright 2017 Google Inc. All Rights Reserved.
    Use of this source code is governed by an MIT-style license that
    can be found in the LICENSE file at http://angular.io/license */

src/app/app.component.ts
import { Component, OnInit, ViewChild } from '@angular/core';
import { MatPaginator } from '@angular/material/paginator';
import { MatSort } from '@angular/material/sort';
import { MatTableDataSource } from '@angular/material/table';

export interface UserData {
  sno: any;
  deviceName: string;
  developer: string;
  download: string;
}

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})

export class AppComponent implements OnInit {
  title = 'msm';

  displayedColumns: string[] = ['sno', 'deviceName', 'developer', 'download'];
  dataSource: MatTableDataSource<UserData>;

  @ViewChild(MatPaginator, { static: true }) paginator: MatPaginator;
  @ViewChild(MatSort, { static: true }) sort: MatSort;

  constructor() {
    // Create 100 users
    const devices = [{
      sno: 1,
      deviceName: 'OP3 & OP3T',
      developer: 'SuperDroidBond',
      download:   "https://sourceforge.net/projects/xtended/files/oneplus3/" 
    },
    {
      sno: 2,
      deviceName: 'davinci',
      developer: '@SVB_22',
      download: 'https://sourceforge.net/projects/xtended/files/davinci/'
    },
    {
      sno: 3,
      deviceName: 'guacamole',
      developer: '@ChandraChaganti',
      download: 'https://sourceforge.net/projects/xtended/files/guacamole/'
    },
    {
      sno: 4,
      deviceName: 'dipper',
      developer: '@DarkAngelGR',
      download: 'https://sourceforge.net/projects/xtended/files/dipper/'
    },
    {
      sno: 5,
      deviceName: 'hltetmo',
      developer: '@Jprimero15',
      download: 'https://sourceforge.net/projects/xtended/files/hltetmo/'
    },
    {
      sno: 6,
      deviceName:	'hltekor',
      developer: '@Jprimero15',
      download: 'https://sourceforge.net/projects/xtended/files/hltekor/'
    },
    {
      sno: 7,
      deviceName: 'hltechn',
      developer: '@Jprimero15',
      download: 'https://sourceforge.net/projects/xtended/files/hltechn/'
    },
    {
      sno: 8,
      deviceName: 'hlte',
      developer: '@Jprimero15',
      download: 'https://sourceforge.net/projects/xtended/files/hlte/'
    },
    {
      sno: 9,
      deviceName: 'whyred',
      developer: '@sreekfreak995',
      download: 'https://sourceforge.net/projects/xtended/files/whyred/'
    },
    {
      sno: 10,
      deviceName: 'mido',
      developer: '@Sagarrokade006',
      download: 'https://sourceforge.net/projects/xtended/files/mido/'
    },
    {
      sno: 11,
      deviceName: 'z2plus',
      developer: '@PptOo7',
      download: 'https://sourceforge.net/projects/xtended/files/z2plus/'
    },
    {
      sno: 12,
      deviceName: 'tissot',
      developer: '@amarbajpai',
      download: 'https://sourceforge.net/projects/xtended/files/tissot/'
    },
    {
      sno: 13,
      deviceName: 'X00T',
      developer: '@pk965',
      download: 'https://sourceforge.net/projects/xtended/files/X00T/'
    },
    {
      sno: 14,
      deviceName: 'cedric',
      developer: '@kiam001',
      download: 'https://sourceforge.net/projects/xtended/files/cedric/'
    },
    {
      sno: 15,
      deviceName: 'violet',
      developer: '@mavji007',
      download: 'https://sourceforge.net/projects/xtended/files/violet/'
    },
    {
      sno: 16,
      deviceName: 'beryllium',
      developer: '@Dev_Mashru',
      download: 'https://sourceforge.net/projects/xtended/files/beryllium/'
    },
    {
      sno: 17,
      deviceName: 'bacon',
      developer: '@xawlw',
      download: 'https://sourceforge.net/projects/xtended/files/bacon/'
    },
    {
      sno: 18,
      deviceName: 'ginkgo',
      developer: '@xawlw',
      download: 'https://sourceforge.net/projects/xtended/files/ginkgo/'
    },
    {
      sno: 19,
      deviceName: 'santoni',
      developer: '@beingstargazer',
      download: 'https://sourceforge.net/projects/xtended/files/santoni/'
    },
    {
      sno: 20,
      deviceName: 'enchilada',
      developer: '@UltraNoob5',
      download: 'https://sourceforge.net/projects/xtended/files/enchilada/'
    },
    {
      sno: 21,
      deviceName: 'fajita',
      developer: '@UltraNoob5',
      download: 'https://sourceforge.net/projects/xtended/files/fajita/'
    },
    {
      sno: 22,
      deviceName: 'sanders',
      developer: '@saisamy95',
      download: 'https://sourceforge.net/projects/xtended/files/sanders/'
    },
    {
      sno: 23,
      deviceName: 'vince',
      developer: '@MahmoudK1000',
      download: 'https://sourceforge.net/projects/xtended/files/vince/'
    },
    {
      sno: 24,
      deviceName: 'wayne',
      developer: '@AmyButDark',
      download: 'https://sourceforge.net/projects/xtended/files/wayne/'
    },
    {
      sno: 25,
      deviceName: 'rosy',
      developer: '@Speedhorn',
      download: 'https://sourceforge.net/projects/xtended/files/rosy/'
    },
    {
      sno: 26,
      deviceName: 'X01BD',
      developer: '@SonalSingh18',
      download: 'https://sourceforge.net/projects/xtended/files/X01BD/'
    },
    {
      sno: 27,
      deviceName: 'onclite',
      developer: '@mylove90',
      download: 'https://sourceforge.net/projects/xtended/files/onclite/'
    },
    {
      sno: 28,
      deviceName: 'lavender',
      developer: '@Hazama25',
      download: 'https://sourceforge.net/projects/xtended/files/lavender/'
    },
    {
      sno: 29,
      deviceName: 'lucye-us997',
      developer: '@wolfsvk',
      download: 'https://sourceforge.net/projects/xtended/files/lucye/US997/'
    },
    {
      sno: 30,
      deviceName: 'lucye-h872',
      developer: '@wolfsvk',
      download: 'https://sourceforge.net/projects/xtended/files/lucye/H872/'
    },
    {
      sno: 31,
      deviceName: 'lucye-h870',
      developer: '@wolfsvk',
      download: 'https://sourceforge.net/projects/xtended/files/lucye/H870/'
    },
    {
      sno: 32,
      deviceName: 'payton',
      developer: '@cool585',
      download: 'https://sourceforge.net/projects/xtended/files/payton/'
    },
    {
      sno: 33,
      deviceName: 'hotdogb',
      developer: '@mukesh22584',
      download: 'https://sourceforge.net/projects/xtended/files/hotdogb/'
    },
    {
      sno: 34,
      deviceName: 'oneplus3',
      developer: '@SuperDroidBond',
      download: 'https://sourceforge.net/projects/xtended/files/oneplus3/'
    },
  ];

    // Assign the data to the data source for the table to render
    this.dataSource = new MatTableDataSource(devices);
  }

  ngOnInit() {
    this.dataSource.paginator = this.paginator;
    this.dataSource.sort = this.sort;
  }

  applyFilter(event: Event) {
    const filterValue = (event.target as HTMLInputElement).value;
    this.dataSource.filter = filterValue.trim().toLowerCase();
    if (this.dataSource.paginator) {
      this.dataSource.paginator.firstPage();
    }
  }
}
